hs-bindgen-runtime/test/Test/HsBindgen/Runtime/SizedByteArray.hs

```diff
@@ -0,0 +1,225 @@
+            , testProperty "indexOffAddr matches readOffAddr" prop_indexOffAddr_consistency
+            ]
+        , testGroup "PrimArray integration" [
+              testCase "create and index PrimArray" test_primArray_basic
+            , testProperty "PrimArray roundtrip" prop_primArray_roundtrip
+            ]
+        , testGroup "Size and alignment" [
+              testCase "sizeOf# matches type parameter" test_sizeOf
+            , testCase "alignment# matches type parameter" test_alignment
+            ]
+        ]
+    ]
+
+{-------------------------------------------------------------------------------
+  Helper: Create SizedByteArray from bytes
+-------------------------------------------------------------------------------}
+
+-- Create a SizedByteArray from a list of bytes
+mkSizedByteArray :: forall n m. (KnownNat n, KnownNat m) => [Word8] -> SizedByteArray n m
+mkSizedByteArray bytes =
+    SizedByteArray $ BA.runByteArray $ do
+        let n = sizeOf (undefined :: SizedByteArray n m)
+        arr <- BA.newByteArray n
+        mapM_ (uncurry (BA.writeByteArray arr)) (zip [0..] bytes)
+        return arr
+
+-- Extract bytes from SizedByteArray
+toBytes :: forall n m. (KnownNat n, KnownNat m) => SizedByteArray n m -> [Word8]
+toBytes (SizedByteArray (ByteArray ba)) =
+    let n = sizeOf (undefined :: SizedByteArray n m)
+    in map (\i -> BA.indexByteArray (ByteArray ba) i) [0..n-1]
+
+{-------------------------------------------------------------------------------
+  ByteArray operations tests
+-------------------------------------------------------------------------------}
+
+test_indexByteArray_single :: IO ()
+test_indexByteArray_single = do
+    -- Create a ByteArray containing one SizedByteArray of 4 bytes
+    let input = mkSizedByteArray @4 @4 [0x12, 0x34, 0x56, 0x78]
+
+    -- Create a PrimArray with one element
+    let arr = PA.primArrayFromList [input]
+    let result = PA.indexPrimArray arr 0
+
+    toBytes result @?= [0x12, 0x34, 0x56, 0x78]
+
+test_indexByteArray_multiple :: IO ()
+test_indexByteArray_multiple = do
+    -- Create three SizedByteArrays
+    let input1 = mkSizedByteArray @4 @4 [0x11, 0x22, 0x33, 0x44]
+    let input2 = mkSizedByteArray @4 @4 [0x55, 0x66, 0x77, 0x88]
+    let input3 = mkSizedByteArray @4 @4 [0xAA, 0xBB, 0xCC, 0xDD]
+
+    -- Create a PrimArray with three elements
+    let arr = PA.primArrayFromList [input1, input2, input3]
+
+    toBytes (PA.indexPrimArray arr 0) @?= [0x11, 0x22, 0x33, 0x44]
+    toBytes (PA.indexPrimArray arr 1) @?= [0x55, 0x66, 0x77, 0x88]
+    toBytes (PA.indexPrimArray arr 2) @?= [0xAA, 0xBB, 0xCC, 0xDD]
+
+prop_readByteArray_roundtrip :: [Word8] -> Property
+prop_readByteArray_roundtrip bytes = ioProperty $ do
+    let input = mkSizedByteArray @8 @8 (take 8 $ bytes ++ repeat 0)
+
+    -- Create a mutable array, write the value, read it back
+    marr <- BA.newByteArray 8
+    BA.writeByteArray marr 0 input
```

Comment:
I'm happy you added tests for all this! Since you've gone through all that trouble anyway, and since these are already _properties_, not just unit tests, can we test with indices other than 0?